conn.go

```diff
@@ -176,7 +165,10 @@ func (c *Conn) SendMessages(messages []Message) ([]Message, error) {
 			d.debugf(1, "send msgs: err: %v", err)
 		})
 
-		return nil, err
+		return nil, &OpError{
+			Op:  "send-messages",
+			Err: err,
```

Comment:
I'm leaving this comment here for lack of a better place. The high level operation is `"send-messages"`, but we know nothing about `err` itself. Presumably, the implementation of `c.sock` has made a system call, and the system call has returned an error. For completeness, I think the system call (e.g. `sendmsg` ) should be recorded in the form of an `*os.SyscallError`, and this should happen inside the `Socket` implementation.

The code right here should not change, but I think all the code that is making system calls should.

Reply:
SGTM!

Reply:
This has been applied, review when ready!

Reply:
Looks good. Marking as resolved.